orchestrator/src/compression/blob.rs

```diff
@@ -0,0 +1,303 @@
+    let total_contracts = len_storage_diffs + leftover_addresses.len();
+    blob_data[0] = Felt::from(total_contracts);
+    tracing::debug!("Total contract updates in blob: {}", total_contracts);
+
+    // Add declared classes count
+    blob_data.push(Felt::from(state_diff.declared_classes.len()));
+
+    // Process each declared class
+    for DeclaredClassItem { class_hash, compiled_class_hash } in state_diff.declared_classes.into_iter() {
+        blob_data.push(class_hash);
+        blob_data.push(compiled_class_hash);
+    }
+
+    tracing::debug!("Created blob data with {} elements", blob_data.len());
+
+    Ok(blob_data)
+}
+
+/// Creates a DA word with information about a contract
+///
+/// # Arguments
+/// * `class_flag` - Indicates if a new class hash is present
+/// * `nonce_change` 0 Optional nonce value as a Felt
+/// * `num_change` - Number of storage updates
+/// * `version` - Version string to determine the encoding format
+///
+/// # Returns
+/// A `Felt` representing the encoded DA word
+pub fn da_word(
+    class_flag: bool,
+    nonce_change: Option<Felt>,
+    num_changes: u64,
+    version: StarknetVersion,
+) -> Result<Felt, JobError> {
+    // Parse version to determine format
+    let is_gte_v0_13_3 = version >= StarknetVersion::V0_13_3;
+    // let mut da_word = Felt::ZERO;
+
+    let mut da_word = BigUint::zero();
+
+    if is_gte_v0_13_3 {
+        // v0.13.3+ format:
+        //
+        // Binary encoding layout (252 bits total):
+        //
+        // Bit positions:
+        // 251                                                                    0
+        // |                                                                      |
+        // v                                                                      v
+        // ┌───────────────┬─────────────────────────────────────┬───┬───┬────────┐
+        // │   new_nonce   │           n_updates                 │n_u│cls│ Zeros  │
+        // │   (64 bits)   │        (8 or 64 bits)               │len│flg│        │
+        // │   [251:188]   │     [187:124] or [187:180]          │[1]│[0]│        │
+        // └───────────────┴─────────────────────────────────────┴───┴───┴────────┘
+        //
+        // Field breakdown:
+        // - new_nonce (64 bits)     : [251:188] - Nonce value
+        // - n_updates (variable)    : [187:124] (64-bit) or [187:180] (8-bit, then padded by 0s) - Number of updates
+        // - n_updates_len (1 bit)   : [1] - Length indicator (1=8-bit, 0=64-bit n_updates)
+        // - class_flag (1 bit)      : [0] - Class changed or not
+        //
+        // Total used bits: 64 + (8|64) + 1 + 1 = 74 or 130 bits
+        // Remaining bits: 252 - (74|130) = 178 or 122 bits (reserved/padding)
+        //
+
+        // Add new_nonce (64 bits)
+        if let Some(new_nonce) = nonce_change {
+            let new_nonce = new_nonce.to_u64().expect("nonce too big");
```

Comment:
let's throw a proper error here:

```suggestion
            let new_nonce = new_nonce.to_u64()
    .ok_or_else(|| JobError::Other(OtherError(eyre!("Nonce value {} exceeds u64 maximum", new_nonce))))?;
```